packages/core/src/signer/wallet-client.ts

```diff
@@ -0,0 +1,40 @@
+import { getAddress, type ByteArray, type Hex, type WalletClient } from "viem";
+import type { SmartAccountSigner } from "./types";
+import type { SignTypedDataParams } from "../account/types";
+
+export class WalletClientSigner implements SmartAccountSigner {
+  private client: WalletClient;
+
+  constructor(client: WalletClient) {
+    this.client = client;
+  }
+
+  getAddress: () => Promise<`0x${string}`> = async () => {
+    let addresses = await this.client.getAddresses();
+    return getAddress(addresses[0]);
+  };
+
+  readonly signMessage: (
+    message: string | Hex | ByteArray
+  ) => Promise<`0x${string}`> = async (message) => {
+    if (typeof message === "string" && !message.startsWith("0x")) {
```

Comment:
you can probably fix the lint error by doing
```suggestion
    if (typeof message === "string" && !isHex(message)) {
```

isHex is exported by viem. that was my bad, totally forgot that helper existed